synapse/lib/types.py

```diff
@@ -551,6 +551,11 @@ def _normPyTuple(self, valu):
             if _type.ismutable:
                 self.ismutable = True
 
+            if _type.deprecated:
```

Comment:
This check (and the ismutable one above actually) seems like they should be done during init, not during norming.